flink/src/main/java/org/apache/iceberg/flink/FlinkFilters.java

```diff
@@ -0,0 +1,266 @@
+    if (op != null) {
+      switch (op) {
+        case IS_NULL:
+          Optional<String> name = toReference(singleton(call, FieldReferenceExpression.class).orElse(null));
+          return name.map(Expressions::isNull);
+
+        case NOT_NULL:
+          Optional<String> nameNotNull = toReference(singleton(call, FieldReferenceExpression.class).orElse(null));
+          return nameNotNull.map(Expressions::notNull);
+
+        case LT:
+          return convertComparisonExpression(Expressions::lessThan, Expressions::greaterThan, call);
+
+        case LT_EQ:
+          return convertComparisonExpression(Expressions::lessThanOrEqual, Expressions::greaterThanOrEqual, call);
+
+        case GT:
+          return convertComparisonExpression(Expressions::greaterThan, Expressions::lessThan, call);
+
+        case GT_EQ:
+          return convertComparisonExpression(Expressions::greaterThanOrEqual, Expressions::lessThanOrEqual, call);
+
+        case EQ:
+          return handleNaN(Expressions::equal, Expressions::isNaN, call);
+
+        case NOT_EQ:
+          return handleNaN(Expressions::notEqual, Expressions::notNaN, call);
+
+        case NOT:
+          Optional<Expression> child = convert(singleton(call, CallExpression.class).orElse(null));
+          return child.map(Expressions::not);
+
+        case AND:
+          return convertLogicExpression(Expressions::and, call);
+
+        case OR:
+          return convertLogicExpression(Expressions::or, call);
+
+        case STARTS_WITH:
+          return convertLike(call);
+      }
+    }
+
+    return Optional.empty();
+  }
+
+  private static <T extends ResolvedExpression> Optional<T> singleton(CallExpression call,
+                                                                      Class<T> expectedChildClass) {
+    List<ResolvedExpression> children = call.getResolvedChildren();
+    if (children.size() != 1) {
+      return Optional.empty();
+    }
+
+    ResolvedExpression child = children.get(0);
+    if (!expectedChildClass.isInstance(child)) {
+      return Optional.empty();
+    }
+
+    return Optional.of(expectedChildClass.cast(child));
+  }
+
+  private static Optional<Expression> convertLike(CallExpression call) {
+    Tuple2<String, Object> tuple2 = parseFieldAndLiteral(call);
+    if (tuple2 == null) {
+      return Optional.empty();
+    }
+
+    String pattern = tuple2.f1.toString();
+    Matcher matcher = STARTS_WITH_PATTERN.matcher(pattern);
+
+    // exclude special char of LIKE
+    // '_' is the wildcard of the SQL LIKE
+    if (!pattern.contains("_") && matcher.matches()) {
+      return Optional.of(Expressions.startsWith(tuple2.f0, matcher.group(1)));
+    }
+
+    return Optional.empty();
+  }
+
+  private static Optional<Expression> convertLogicExpression(BiFunction<Expression, Expression, Expression> function,
+                                                             CallExpression call) {
+    List<ResolvedExpression> args = call.getResolvedChildren();
```

Comment:
If this is going to access children from this list directly using `get(1)` then it should also check that there are exactly 2 children.

Reply:
If we enter this method, it means that it is a legal logic expression,  if we add a judgment, it is indeed safer